### DISCUSSION TITLE
Field cannot autocompletion when its a SQLModel

### DISCUSSION THREAD

**Discussion by zhaowcheng** [fastapi/sqlmodel#866] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the SQLModel documentation, with the integrated search.
- [X] I already searched in Google "How to X in SQLModel" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to SQLModel but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to SQLModel but to [SQLAlchemy](https://github.com/sqlalchemy/sqlalchemy).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from sqlmodel import SQLModel
from typing import Optional

class ModelNode(SQLModel):
    id: int
    name: str

class Node:
    id: int
    name: str

class Cluster(SQLModel):
    model_node: Optional[ModelNode] = None
    node: Optional[Node] = None
```

### Description

Cluster.node. can autocompletion, but Cluster.model_node. cannot

### Operating System

macOS

### Operating System Details

IDE: vscode

### SQLModel Version

0.0.16

### Python Version

3.10.5

### Additional Context

![image](https://github.com/tiangolo/sqlmodel/assets/4382523/48c09870-e5e7-4322-85c3-c326765a7bc7)

![image](https://github.com/tiangolo/sqlmodel/assets/4382523/5e9f25fd-b47e-41bc-9e2e-1a783f506f03)

**Comment by YuriiMotov** [comment]
Works fine in current version:
<img width="622" height="476" alt="image" src="https://github.com/user-attachments/assets/479a5abf-21b3-482e-b90c-8074e48fb9a0" />
